src/main.rs

```diff
@@ -141,35 +140,55 @@ async fn main() -> eyre::Result<()> {
 
     match cli.command {
         Command::Init { lib, api, package } => {
-            let r#type = if lib {
+            let kind = if lib {
                 PackageType::Lib
             } else if api {
                 PackageType::Api
             } else {
                 PackageType::Impl
             };
 
-            command::init(r#type, package).await
+            command::init(kind, package)
+                .await
+                .wrap_err(miette!("init command failed"))
         }
         Command::Add {
             registry,
             dependency,
-        } => command::add(registry, &dependency).await,
-        Command::Remove { package } => command::remove(package).await,
+        } => command::add(registry, &dependency)
+            .await
+            .wrap_err(miette!("add command failed")),
+        Command::Remove { package } => command::remove(package)
+            .await
+            .wrap_err(miette!("remove command failed")),
```

Comment:
these are still bad error messages. It would be neat to do something meaningful here like `failed to remove `{package}` from `Proto.toml`` - smth like this should be the toplevel message for all of the commands

Reply:
I replied to your earlier comment - this is actually very hard with the current architecture, we'd need to either return a structured error from the command or move package name resolution logic to `main.rs`. I prefer to not attempt that in this PR, it's grown twice as large as originally intended already.

Reply:
Huh? How is it hard?

You can just change the message:

```suggestion
            .wrap_err(miette!("failed to remove `{package}` from `{MANIFEST_FILE}`")),
```

Reply:
For this one command, yes, we know the package name because it was passed as an argument, but for many others that is not available at this level. So this is hard:

> smth like this should be the toplevel message for all of the commands

Reply:
#129